### ISSUE TITLE
Investigate insufficiencies in migration script for Quarkus 3

### ISSUE THREAD

**Issue by mabartos** [keycloak/keycloak#16247]
### Description

For migration from Quarkus 2.x to Quarkus 3.x, the Quarkus team provides a migration script, which should automatically set up the whole project for Quarkus 3.x.

This should be our starting point for upgrading our project to Quarkus 3. It'd also be good to provide some feedback to the Quarkus team about the overall migration of our project; as Keycloak is one of the biggest projects, they might be curious about the migration process.

**Comment by mabartos** [comment]
### Fundamental findings about the migration script
The migration script upgrades all dependencies and relevant packages in the codebase from using JavaEE to JakartaEE. 
However, we've encountered an issue with the script such as the packages in the codebase are overridden from `javax.` to `jakarta.`, but related dependencies are not upgraded/substituted. 

#### Missing conversions for dependencies included in Keycloak
Syntax: groupId(artifactId)

- com.sun.activation(jakarta.activation) -> jakarta.activation(jakarta.activation-api)
- org.jboss.spec.javax.ws.rs(jboss-jaxrs-api_2.1_spec) -> jakarta.ws.rs(jakarta.ws.rs-api)
- org.jboss.spec.javax.transaction(jboss-transaction-api_1.3_spec) -> jakarta.transaction(jakarta.transaction-api)

**Comment by mabartos** [comment]
@ia3andy Hi! I see that you're the main contributor related to Quarkus 3 migration[1], so you probably have some answers :)) 

We've tried to execute the migration script(openRewrite recipe)[2] for Quarkus 3 upgrade, and we've found some issues in our project (described above). 

Do you think it's relevant to include mentioned conversions above also to the migration script? I think the conversions from `com.sun.*` to `jakarta.*` makes perfect sense. However, I'm not entirely sure about the `org.jboss.spec.javax.*` to `jakarta.*` in terms of possible incompatibilities; I'm not aware of any right now. 

Do you have any objections to including them in the migration script?

[1] https://github.com/quarkusio/quarkus/issues/29000 
[2] https://raw.githubusercontent.com/quarkusio/quarkus/main/jakarta/quarkus3.yml

**Comment by ia3andy** [comment]
@maxandersen @gsmet this should be integrated in the script right?

**Comment by maxandersen** [comment]
First intuition is that it makes sense to add it. But Make PRs for the various parts and easier to comment/review.

**Comment by gsmet** [comment]
Yes, we should include them. The OpenRewrite recipe we use for Quarkus can be a good guide for that.

**Comment by mabartos** [comment]
The investigation was done and details described in the comment https://github.com/keycloak/keycloak/issues/16247#issuecomment-1370978695
